src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -903,10 +906,15 @@ private ProjectInstance CreateTraversalInstance(string wrapperProjectToolsVersio
                 _submissionId
                 );
 
-            // Make way for the real ones                
-            foreach (string targetName in _defaultTargetNames)
+            // Make way for the real ones
+            foreach (string targetName in dummyTargetsForEvaluationTime)
             {
-                traversalInstance.RemoveTarget(targetName);
+                // Remove targets only if they were the dummy ones (from the metaproj path),
+                // but leave them if they're from another source (imported/overridden).
+                if (traversalInstance.Targets[targetName].Location.File == traversalProject.FullPath)
```

Comment:
What if they're the dummy ones but there's before / after targets hooked to them? Still good to remove?

Reply:
Talked offline. Target execution uses the object model only, luckily.